### DISCUSSION TITLE
HL7 enricher with GET

### DISCUSSION THREAD

**Discussion by vincentstm33** [nextgenhealthcare/connect#5065] [Q&A]
Hi, 

i'm newbe on Mirth.
I am trying to enricher an HL7 message with multiple GET.
To do this, i have created a Channel.
In the source, i put "channel reader", and in the destination, i use multiple GET with HTTP.
In edit transformer, i use a javascript :

var url = "http://hapi.fhir.org/baseR4/Patient/" + msg['PID']['PID.3']['PID.3.1'].toString() + "/_history/1?_pretty=true";
channelMap.put("PatientID", msg['PID']['PID.3']['PID.3.1'].toString());
channelMap.put("URL", url);

I obtain this response (mix betweek xml and json) : 

<HttpResponse>
    <Status>HTTP/1.1 200 OK</Status>
    <Header>
        <Field>
            <Name>Transfer-Encoding</Name>
            <Value>chunked</Value>
        </Field>
        <Field>
            <Name>Server</Name>
            <Value>nginx/1.18.0 (Ubuntu)</Value>
        </Field>
        <Field>
            <Name>ETag</Name>
            <Value>W/"1"</Value>
        </Field>
        <Field>
            <Name>X-Request-ID</Name>
            <Value>EyJ57STWOM</Value>
        </Field>
        <Field>
            <Name>Content-Location</Name>
            <Value>http://hapi.fhir.org/baseR4/Patient/2005433/_history/1</Value>
        </Field>
        <Field>
            <Name>Connection</Name>
            <Value>keep-alive</Value>
        </Field>
        <Field>
            <Name>Last-Modified</Name>
            <Value>Mon, 12 Apr 2021 03:03:50 GMT</Value>
        </Field>
        <Field>
            <Name>Date</Name>
            <Value>Mon, 28 Feb 2022 17:35:55 GMT</Value>
        </Field>
        <Field>
            <Name>Content-Type</Name>
            <Value>application/fhir+json;charset=utf-8</Value>
        </Field>
        <Field>
            <Name>X-Powered-By</Name>
            <Value>HAPI FHIR 5.7.0-PRE9-SNAPSHOT/76c37458b3/2022-01-30 REST Server (FHIR Server; FHIR 4.0.1/R4)</Value>
        </Field>
    </Header>
    <Body multipart="no">{
  "resourceType": "Patient",
  "id": "2005433",
  "meta": {
    "versionId": "1",
    "lastUpdated": "2021-04-12T03:03:50.817+00:00",
    "source": "#JFIL3prPb5l0Arv5"
  },
  "text": {
    "status": "generated",
    "div": "&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;First Lastname1 Lastname2&lt;/div&gt;"
  },
  "identifier": [ {
    "system": "http://clinfhir.com/fhir/NamingSystem/identifier",
    "value": "first.last"
  } ],
  "name": [ {
    "use": "official",
    "text": "First Lastname1 Lastname2",
    "family": "Lastname1 Lastname2",
    "given": [ "First" ]
  } ],
  "gender": "male",
  "birthDate": "2021-04-11"
}</Body>
</HttpResponse>

I would like to convert it with my initial message.
I would like to make name family in PID-5 for example.
I tried with menu "edit response" but without success.

Please, can you help me?

**Comment by pacmano1** [comment]
Please read https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.   Also see https://github.com/nextgenhealthcare/connect/discussions/4824.   Please format your code to make it easier for folks to read.

Although this is not the mirth forums, please  also read: https://forums.mirthproject.io/articles/177082-forum-etiquette, please note bullet 7, 8, 9,10, 11.

In your case exact screenshots for the response to your call would be helpful as would re-formatting your original post.

Effectively in the response transformer of the http call you will interpret the response and build/change/construct the message as needed.   As near as I can tell, that is a standard JSON response:
```json
{
    "resourceType": "Patient",
    "id": "2005433",
    "meta": {
        "versionId": "1",
        "lastUpdated": "2021-04-12T03:03:50.817+00:00",
        "source": "#JFIL3prPb5l0Arv5"
    },
    "text": {
        "status": "generated",
        "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">First Lastname1 Lastname2</div>"
    },
    "identifier": [{
        "system": "http://clinfhir.com/fhir/NamingSystem/identifier",
        "value": "first.last"
    }],
    "name": [{
        "use": "official",
        "text": "First Lastname1 Lastname2",
        "family": "THIS IS THE LASTNAME",
        "given": ["First"]
    }],
    "gender": "male",
    "birthDate": "2021-04-11"
}
```

So in a response transformer, the field in the above that has `THIS IS THE LASTNAME` is `msg['name'][0]['family']`
